### ISSUE TITLE
Average DNS response time worsened

### ISSUE THREAD

**Issue by ynkhr** [AdguardTeam/AdguardForiOS#2161]
<Premise: Environment used>
iPhone 12 Pro Max (16.4.1) / iPad Pro (16.4.1)
　The events described below occurred with 16.4.x, rather than with the earlier version 16.3.x, etc,
　　The events described below did not occur with 16.4.x, but rather with earlier versions, such as 16.3.x.
AdGuard Pro: AdGuard For iOS 4.4.5 (990)
Safari blocking:
　　a) AdGuard Base Filter
　　b) AdGuard Mobile Ad Filter
　　c) AdGuard Japanese filter
DNS Filters (DNS servers and filter availability):
　　a) System default (filter: AdGuard DNS filter)
　　b) AdGuard DNS (no filter)
　　c) Cloudflare DNS (filter: AdGuard DNS filter)

<What is happening and what we would like to ask you>
1) Deterioration of average response time
For the past 2 to 3 months, the average response time of Adguard Pro has been slow, more than 1.0s.
The response time has been even worse, especially in the last 2 to 3 days, and we have never seen such figures as 40s to 54s.
The above may be the cause of the slowdown in web site display.
I remember that it used to be 7~8ms at the earliest and 20~30ms at the latest.
　Can it be speeded up by reviewing some settings or usage?
　
2) Worsening of filter update time
Even if you try to update the filter under Wi-Fi connection, it takes more than 1 minute to update the filter.
　(It used to take at least 10 seconds...). (I think it used to be 10+ seconds at most...)
Is it possible to shorten the filter update time?
　Does it depend on the hardware specs?

**Comment by EZiskpsy05** [comment]
I think you should clear the cache, remove the filter lists then add them again. Try to clear your DNS cache and refresh you Wi-Fi too.

**Comment by ynkhr** [comment]
Thank you for your confirmation.
Can you please tell us how to clear the DNS cache?
I have performed the following three steps and the situation you reported is happening.
- Restarting the iPhone
- Reset network settings
- Clear Safari cache

Update interval is set to 24h.
No change after clearing Adguard stats.

**Comment by EZiskpsy05** [comment]
For 1)
It seems that there is a problem with the Adguard Pro software that is causing slow response times. You should:
- Update Adguard Pro
- Check your settings (you may want to check if any filters are causing unnecessary delays or if the software is configured to block too many requests)
- If possible, change your DNS to a different DNS provider.
If you have tried resetting your network state and done all the steps and it still does not work, you may want to contact Adguard support.
For 2)
You can visit this topic to learn more about updating filter lists https://github.com/AdguardTeam/AdguardForiOS/issues/1844

**Comment by ynkhr** [comment]
Thank you for your reply.

1)
I am using AdGuard Pro 4.4.6 (998) and am aware of the latest version available for download from the AppStore.

As for the settings, I have turned on the following for the filter based on the advice from support.
When I first contacted support, I had c) turned off and the "AdGuard Mobile Ads filter" turned on.
a)AdGuard Base filter
b) AdGuard Tracking Protection filter
c)AdGuard Annoyances filter
d)AdGuard Japanese filter

We have tried all of the DNS providers, but the problem has not changed.

We have reset the network settings as we told you earlier.

After contacting support, I was advised to contact them via GitHub.
I will contact support again.

2)
I understand that with iOS, it is inevitable that it takes time.

**Comment by Versty** [comment]
@EZiskpsy05 Hi! Thank you for the details provided. Have you tried to reinstall the application? 

We are also aware of the fact that the time to update the filters is currently too long, the fix is planned.

**Comment by ynkhr** [comment]
Thanks for the reply.

I have tried reinstalling the application.
No change at all.

I would appreciate a reduction in the filter update time.
More importantly, I would like to see an improvement in the average time that is also displayed on the dashboard of the application.
After all, it is not only a display issue on the dashboard, but I find the browser and app communication to be slow.

Thank you in advance for your patience.

**Comment by Versty** [comment]
@ynkhr Does the issue persists on both cellular and wi-fi networks?

We would like you to try DNS from different providers and with different protocols from [this list](https://adguard-dns.io/kb/general/dns-providers/#adguard-dns). Please add the DNS as custom, then check if the issue persists. 

It would be perfect if you start from AdGuard DNS, IPv4: `94.140.15.15` and IIJ.JP DNS: `https://public.dns.iij.jp/dns-query`

**Comment by ynkhr** [comment]
Thank you for your reply.

> @ynkhr Does the issue persists on both cellular and wi-fi networks?
I will answer this point first.
The situation is Yes.

We will try DNS after checking the list.

**Comment by ynkhr** [comment]
For the time being, we have designated the following two DNS servers as recommended by you.
AdGuard DNS (IPv4)
IIJ.JP DNS

As a result, it is the same as the system default.
The average time on the dashboard was about 11.6s.
(In the worst case, it was over 30s.)

The time did not change no matter which DNS server was selected.

We would appreciate it if you could confirm this for us.

**Comment by Versty** [comment]
@ynkhr Got it. We would like to take a look into the logs from your device. 

Please follow the steps below to collect and send them properly:

1. Open Settings --> General --> Turn Advanced mode on --> Advanced settings --> Turn Debug logs switch on;
2. Select the problematic DNS and surf some sites;
3. Disable DNS protection module
4. Open the settings --> Support --> Export logs;
5. System popup will be opened --> Choose "Mail" app;
6. Enter apple@adguard.com as a recipient;
7. Mention this issue number `2161` in the subject and add the time the issue was reproduced at.

**Comment by ynkhr** [comment]
We have not been able to tell you about this on GitHub, but the log was emailed to you on May 7, 2023 at 19:30.

Please take a moment to check it.

**Comment by Versty** [comment]
@ynkhr Thank you for the logs, we are investigating this issue.

**Comment by Versty** [comment]
@ynkhr We see an issue in the logs, your DNS queries are timed out for some reason. 
For a further investigation we would like you to add and select https://dns.google/dns-query DNS server, then collect the logs once again. Please follow the same steps from the [comment](https://github.com/AdguardTeam/AdguardForiOS/issues/2161#issuecomment-1535819043) above.

**Comment by ynkhr** [comment]
Sorry for the delay.
We have set up the DNS you specified and logged again.

We have sent you the logs in a separate email.
(Subject: Issue number 2161)

Thank you for your patience.

2023年5月22日(月) 18:59 Diana ***@***.***>:

**Comment by ynkhr** [comment]
Sorry for the delay.
We have set up the DNS you specified and logged again.

We have sent you the log by e-mail.
(Subject: Issue number 2161)

Thank you for your patience.

**Comment by EZiskpsy05** [comment]
Oh, by the way, check if your company's (or your network's) firewall/protection configuration blocks or denies any changes to the DNS. For example, I set the network configuration in my network system to force/change the DNS setting to use the native DNS (or school network). This conflicts with the DNS protection in Adguard. I also experience some slowness during the update time, but not too much.

**Comment by ynkhr** [comment]
Hello,

Thank you for contacting us.
We believe that Firewall/Protection is not a problem.
Because,
I do not have any special blocking or other settings at home.
I am not connected to the company network.
And the problem is also occurring on public networks such as 4G/5G of
telecommunication carriers.

The logs we sent you were obtained at home so that they match the
environment as much as possible.

If you have any other concerns, we would appreciate it if you could let us
know.

Thank you in advance.

2023年5月26日(金) 21:35 Howard Tran ***@***.***>:

**Comment by ynkhr** [comment]
Hello,
Have you learned anything since then?

**Comment by ynkhr** [comment]
Today I updated Adguard Pro to ver 4.5.
When I did so, the average response time item was changed to the data saved.

Is the slow response time that you have been asking about not quantitatively in the application, but qualitatively?

**Comment by Versty** [comment]
@ynkhr Yes, we found the root cause, thanks to the logs from your device. 
This is clearly a bug. Will be fixed in the future versions.

**Comment by ynkhr** [comment]
Thank you for contacting us.
This event was a bug.
We are glad that we were able to identify the cause.
We look forward to releasing an updated version.

**Comment by ynkhr** [comment]
Has this inquiry been refurbished?
If not, do you have an estimate of when it will be renovated?

**Comment by ynkhr** [comment]
Could you please tell us what the situation is in this case?

**Comment by MadeByPP** [comment]
There is a tag indicating that this issue is scheduled to be fixed in the future version 4.6.

Usually, feature updates are made once a year (version y of x.y.z).

Since the 4.5 update was about 7 months ago, I think there are about 5 more months left until this problem is fixed.

**Comment by ynkhr** [comment]
Thank you for contacting us.
I was just wondering what the status of the response was.

We appreciate the notice and look forward to the resolution of the problem.
However, I am not sure if you will be able to recognize that the problem
has been resolved, since the screen changed after the recent upgrade and
the response time is no longer displayed.

Also, as I think I have told you in the past, the filter update takes a
long time - several minutes even when connected to Wi-Fi.
We hope this issue will be resolved soon as well.

Thank you in advance for your continued support.

2024年3月2日(土) 6:57 P.P ***@***.***>:
